mteb/abstasks/AbsTaskSpeedTask.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+import logging
+import platform
+import time
+from pathlib import Path
+
+import GPUtil
+import numpy as np
+import psutil
+
+from mteb.encoder_interface import Encoder, EncoderWithQueryCorpusEncode
+from mteb.MTEBResults import ScoresDict
+
+from .AbsTask import AbsTask
+
+logger = logging.getLogger(__name__)
+
+
+TOKENS_IN_UGLY_DUCKLING = 3591
+
+
+class AbsTaskSpeedTask(AbsTask):
+    """Abstract class for speed tasks (e.g. CPU, GPU).
+
+    Args:
+        AbsTask (_type_): _description_
+    """
+
+    num_loops = 7
+    device = "cpu"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def load_data(self, **kwargs):
+        """Reads the text 'The Ugly Duckling' as the `test` split with a `text` column."""
+        if self.data_loaded:
+            return
+        file_path = Path(__file__).parent / "the_ugly_duckling.txt"
+        with file_path.open("r") as f:
+            text = f.read()
+        self.dataset = {"test": {"text": text.split("\n\n")}}
+        self.data_loaded = True
+
+    def _get_time_taken(self, model: Encoder, data_split) -> float:
+        start = time.time()
+        model.encode(data_split["text"], device=self.device)
+        time_taken = time.time() - start
+        return time_taken
+
+    def get_system_info(self) -> dict[str, str]:
+        """Returns a dictionary with system information."""
+        info = {}
+        info["platform"] = platform.system()
+        info["platform_release"] = platform.release()
+        info["platform_version"] = platform.version()
+        info["architecture"] = platform.machine()
+        info["processor"] = platform.processor()
+        info["ram"] = str(round(psutil.virtual_memory().total / (1024.0**3))) + " GB"
```

Comment:
Maybe worth explaining in a comment what the math is doing here?

Reply:
Will add a comment explaining conversion from Bytes to GB